agent/api/statechange.go

```diff
@@ -72,6 +73,69 @@ type TaskStateChange struct {
 	Task *Task
 }
 
+// NewTaskStateChangeEvent creates a new task state change event
+func NewTaskStateChangeEvent(task *Task, reason string) (TaskStateChange, error) {
+	var event TaskStateChange
+	taskKnownStatus := task.GetKnownStatus()
+	if !taskKnownStatus.BackendRecognized() {
+		return event, errors.Errorf(
+			"create task state change event api: status not recognized by ECS: %v; task: %s",
+			taskKnownStatus, task.Arn)
+	}
+	if task.GetSentStatus() >= taskKnownStatus {
+		return event, errors.Errorf(
+			"create task state change event api: status [%s] already sent for task %s",
+			taskKnownStatus.String(), task.Arn)
+	}
+
+	event = TaskStateChange{
+		TaskARN: task.Arn,
+		Status:  taskKnownStatus,
+		Reason:  reason,
+		Task:    task,
+	}
+
+	event.SetTaskTimestamps()
+
+	return event, nil
+}
+
+// NewContainerStateChangeEvent creates a new container state change event
+func NewContainerStateChangeEvent(task *Task, cont *Container, reason string) (ContainerStateChange, error) {
+	var event ContainerStateChange
+	contKnownStatus := cont.GetKnownStatus()
+	if !contKnownStatus.ShouldReportToBackend(cont.GetSteadyStateStatus()) {
+		return event, errors.Errorf(
+			"create container state change event api: status not recognized by ECS: %v; task: %s",
+			contKnownStatus, task.Arn)
```

Comment:
We can avoid having the `task.Arn` , `cont.Name` here and in below cases since we are already logging them in catch.